cluster-autoscaler/Dockerfile.autoscaler_juju.amd64

```diff
@@ -0,0 +1,46 @@
+##
+## Build Autoscaler
+##
+FROM golang:1.17.5 AS autoscaler_builder
+ENV GOPATH /gopath/
+ENV PATH $GOPATH/bin:$PATH
+ENV GO111MODULE auto
+
+WORKDIR /app
+
+RUN apt-get update && apt-get --yes install libseccomp-dev
+RUN go version
+RUN go get github.com/tools/godep
+RUN godep version
+COPY . ./
+RUN make build
+
+##
+## Install Juju 
+##
+
+FROM ubuntu:20.04 AS juju_installer
+
+
+WORKDIR /
+
+
+RUN apt-get update && apt-get --yes install curl xz-utils ca-certificates &&\
```

Comment:
can the juju version bits be made as variables? 
```
ENV JUJU_MAJOR 2.9
ENV JUJU_MINOR 22
ENV JUJU_ARCH amd64
ENV JUJU_URL https://launchpad.net/juju/${JUJU_MAJOR}/${JUJU_MAJOR}.${JUJU_MINOR}/+download/juju-${JUJU_MAJOR}.${JUJU_MINOR}-linux-${JUJU_ARCH}.tar.xz

...

    curl  -L0 ${JUJU_URL} && curl -L ${JUJU_URL}/+md5 -o juju.md5 && 
...
    tar xf ./juju-${JUJU_MAJOR}.${JUJU_MINOR}-linux-${JUJU_ARCH}.tar.xz

```

Reply:
I will add in multi-arch support and use build args for the juju versions. As for the main builder, I wasnt too sure I liked their way of doing things, having a multistage build just seemed cleaner.